.github/workflows/build-and-test.yml

```diff
@@ -94,6 +94,10 @@ jobs:
           git clone https://github.com/precice/precice.git precice-core
           mkdir -p precice
           cp precice-core/src/precice/SolverInterface.hpp precice/SolverInterface.hpp
+          cd precice-core
+          mkdir build && cd build
+          cmake .. -DPRECICE_MPICommunication=OFF -DPRECICE_PETScMapping=OFF -DPRECICE_PythonActions=OFF -DBUILD_TESTING=OFF
+          cp src/precice/Version.h ../../precice/
```

Comment:
This will break the moment we add another generated header.

A long-term solution would be to install preCICE and use it from that location.
A even-longer-term solution is to use pkg-config in the setup.py to detect preCICE.

Reply:
Yes it is true that this would break immediately when we add another generated header, but for the mock testing of the python-bindings I need to not have preCICE built on the system, so I am left with the option to either create the header via cmake and copy it over, or delete the include line from `SolverInterface.hpp`, but the later solution will also break with another generated header. Is there a solution for when preCICE is not built?

Reply:
You could use pkg-config for the include-arguments only. This works from the build directory after executing CMake.
Then you can build and provide linker flags to your mocked library.